src/Symfony/Component/ClassLoader/ClassCollectionLoader.php

```diff
@@ -58,7 +58,12 @@ public static function load($classes, $cacheDir, $name, $autoReload, $adaptive =
 
         $classes = array_unique($classes);
 
-        $cache = $cacheDir.'/'.$name.$extension;
+        // cache the core classes
+        if (!is_dir($cacheDir) && !@mkdir($cacheDir, 0777, true) && !is_dir($cacheDir)) {
+            throw new \RuntimeException(sprintf('Class Collection Loader was not able to create directory "%s"', $cacheDir));
+        }
+        $cacheDir = rtrim(realpath($cacheDir), '/'.DIRECTORY_SEPARATOR);
+        $cache = $cacheDir.DIRECTORY_SEPARATOR.$name.$extension;
```

Comment:
It is: these are some lines move from the bottom to the before the below loop.
https://github.com/symfony/symfony/pull/19859/files?w=1